src/pudl/output/eia930.py

```diff
@@ -74,25 +75,93 @@ def _years_from_context(context) -> list[int]:
+        "granularity",
+    ]
+    return pd.concat(
+        [
+            _out_eia930__hourly_subregion_demand[common_cols],
+            _out_eia930__hourly_operations[common_cols],
+        ]
+    )
+
+
+imputed_combined_demand_assets = impute_timeseries_asset_factory(
+    input_asset_name="_out_eia930__combined_demand",
+    output_asset_name="_out_eia930__combined_imputed_demand",
     years_from_context=_years_from_context,
     value_col="demand_reported_mwh",
     imputed_value_col="demand_imputed_pudl_mwh",
-    id_col="combined_subregion_ba_id",
-    settings=ImputeTimeseriesSettings(method_overrides={2019: "tnn", 2025: "tnn"}),
+    id_col="combined_subregion_ba_code_eia",
+    simulation_group_col="granularity",
+    settings=ImputeTimeseriesSettings(
+        simulate_flags_settings=SimulateFlagsSettings(
+            num_months=10, output_io_manager_key="parquet_io_manager"
+        ),
+    ),
+    output_io_manager_key="io_manager",
 )
 
 
-imputed_ba_demand_assets = impute_timeseries_asset_factory(
-    input_asset_name="_out_eia930__hourly_operations",
-    output_asset_name="out_eia930__hourly_operations",
-    years_from_context=_years_from_context,
-    value_col="demand_reported_mwh",
-    imputed_value_col="demand_imputed_pudl_mwh",
-    id_col="balancing_authority_code_eia",
+@multi_asset(
+    outs={
+        "out_eia930__hourly_subregion_demand": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "out_eia930__hourly_operations": AssetOut(io_manager_key="parquet_io_manager"),
+    }
 )
+def split_ba_subregion_demand(
+    _out_eia930__combined_imputed_demand: pd.DataFrame,
+    core_eia930__hourly_subregion_demand: pd.DataFrame,
+    core_eia930__hourly_operations: pd.DataFrame,
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    """Split combined imputed demand into separate BA/subregion tables."""
+    # Merge core asset on imputed output asset to get columns dropped during imputation
+    ba_demand = _out_eia930__combined_imputed_demand[
+        # Just merge BA data so we have a one-one merge
+        _out_eia930__combined_imputed_demand["granularity"] == "ba"
+    ].merge(
+        # Drop reported demand so we don't have duplicates
+        core_eia930__hourly_operations.drop(columns=["demand_reported_mwh"]),
+        on=["datetime_utc", "balancing_authority_code_eia"],
+        validate="one_to_one",
+        how="left",
+    )
+
+    # Repeat with subregion demand
+    subregion_demand = _out_eia930__combined_imputed_demand[
+        _out_eia930__combined_imputed_demand["granularity"] == "subregion"
+    ].merge(
+        core_eia930__hourly_subregion_demand.drop(columns=["demand_reported_mwh"]),
+        on=[
+            "datetime_utc",
+            "balancing_authority_code_eia",
+            "balancing_authority_subregion_code_eia",
+        ],
+        validate="one_to_one",
+        how="left",
+    )
+
+    return subregion_demand, ba_demand
```

Comment:
These should be returned as a tuple of explicit `dagster.Output` objects, to explicitly link the value with the asset name, so we aren't depending on the brittle ordering of values to keep them straight.  And now I see that there are a bunch of other multi-assets in the timeseries cleaning module that also need this.

Reply:
I went ahead and added all these explicit `Output` constructs since it was easy.

Reply:
Thanks, I also removed the type annotations as that was causing a mismatch

Reply:
Ah yeah, whoops. Thanks for catching that.